src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,323 @@
+        int index = input.indexOf('/');
+        // Returns the input string as a list of strings.
+        if (index > -1) {
+            return input.split("/", 2);
+        } else {
+            String[] tempString = { input, "" };
+            return tempString;
+        }
+    }
+
+    /**
+     * Parses an input into an integer. This is used to determine which element in a
+     * list to perform an operation on.
+     * @param input string to be parsed. Must be non-null and a valid integer.
+     * @return The index of the input if input is valid.
+     * @throws InvalidNumberException if input isn't valid.
+     */
+    private static int parseIndex(String input) throws DukeException {
+        try {
+            return Integer.parseInt(input) - 1;
+        } catch (NumberFormatException e) {
+            throw new InvalidNumberException();
+        }
+    }
+
+    /**
+     * Parses the input and returns a DateTimeWrapper. This is a convenience method
+     * for deadlineParser and eventParser that will try to parse the date and time
+     * out of the input.
+     * @param input the input to parse. Must be non-null and a valid date or time or
+     *              both.
+     * @return a DateTimeWrapper that contains the date and time of the input if
+     *         applicable.
+     * @throws IncorrectFormatException if input isn't valid.
+     */
+    private static DateTimeWrapper parseDate(String input) throws IncorrectFormatException {
+        // Input is parsed into two if it contains both a date and time component.
+        String[] parsedInput = parseInput(input);
+        LocalDate date;
+        LocalTime time;
+        String i1 = parsedInput[0].trim();
+        String i2 = parsedInput[1].trim();
+
+        // Matches both halves of the input to the date.
+        if (i1.matches("[0-9]{2}.[0-9]{2}.[0-9]{4}")) {
+            date = LocalDate.parse(i1.substring(6, 10) + "-" + i1.substring(3, 5) + "-" + i1.substring(0, 2));
+        } else if (i2.matches("[0-9]{2}.[0-9]{2}.[0-9]{4}")) {
+            date = LocalDate.parse(i2.substring(6, 10) + "-" + i2.substring(3, 5) + "-" + i2.substring(0, 2));
+        } else if (i1.matches("[0-9]{4}.[0-9]{2}.[0-9]{2}")) {
+            date = LocalDate.parse(i1.substring(0, 4) + "-" + i1.substring(5, 7) + "-" + i1.substring(8, 10));
+        } else if (i2.matches("[0-9]{4}.[0-9]{2}.[0-9]{2}")) {
+            date = LocalDate.parse(i2.substring(0, 4) + "-" + i2.substring(5, 7) + "-" + i2.substring(8, 10));
+        } else {
+            date = null;
+        }
+
+        // Matches both halves of the input to the time.
+        if (i1.matches("[0-9]{2}.[0-9]{2}.[0-9]{2}")) {
+            time = LocalTime.parse(i1.substring(0, 2) + ":" + i1.substring(3, 5) + ":" + i1.substring(6, 8));
+        } else if (i2.matches("[0-9]{2}.[0-9]{2}.[0-9]{2}")) {
+            time = LocalTime.parse(i2.substring(0, 2) + ":" + i2.substring(3, 5) + ":" + i2.substring(6, 8));
```

Comment:
this method has many if-else statements and is >30 lines of code for one method, perhaps the method can be shortened 😄